### ISSUE TITLE
[DISTANCE] [Tracking] LOW: Refactor UpdateDistanceRequest in Auth/PHP/App to follow 1:1:1 pattern and split it out per field

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#28358]
# Problem
The API command `UpdateDistanceRequest` in PHP was created to take the entire transaction object and validate/save it. This lead to the UI re-using the same command for multiple actions: updating amount, updating description, updating category, etc.. This is bad because when editing the description, the API command expects the entire object to be passed and be valid. If there are invalid waypoints, for example, you cannot update the description. This leads to lots of front-end hacks to get around issues related to errors.

# Solution
Split the command out into field specific commands:
- [x] #29993
- [x] #32316
- [x] #32315
- [x] #32314
- [x] #32313
- [x] #32312
- [x] #32311
- [x] #32310
- [ ] UpdateRequestReceipt (? maybe not)

cc @neil-marcellini @Gonals 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010ea3e1f42f2fd6dd</li>
        <li>Upwork Job ID: 1707222752004390912</li>
        <li>Last Price Increase: 2023-09-28</li>
    </ul>
</details>

**Comment by tgolen** [comment]
@allroundexperts There won't be anything for you to do on this for a while since I will be working on the server APIs. Once those are done, I'll have to implement it in the UI, which I will then need help on and I'll get it auto-assigned at that time.

@kevinksullivan for the same reason, I won't need you assigned to this for now either. Can you please remove the upwork job?

**Comment by kevinksullivan** [comment]
Closed the job 👍

**Comment by tgolen** [comment]
I'm going to split these into individual issues as I work on them. Each field is going to need an Auth PR, a Web-E PR, and an App PR. First issue is https://github.com/Expensify/App/issues/29993 for date.

**Comment by tgolen** [comment]
After discussing this with @cead22 [in Slack](https://expensify.slack.com/archives/C03TQ48KC/p1697751589540909) we decided to narrow the scope of this refactoring to PHP only and to leave the Auth command as it is. That should work. It will mean that we need to call Auth to get the full transaction first, then modify the single field that the frontend is updating, and pass the entire transaction to `EditMoneyRequest`.

**Comment by tgolen** [comment]
OK, nevermind on that last update. Previous convos with @flodnv and @iwiznia made it a little more clear that we should probably do this all the way down in Auth, so I'm going to stick with that original plan. I should be able to get back working on this on Monday.

**Comment by tgolen** [comment]
Split out all the tasks into their own issues today. The first task has 1 PR merged (should be merged tomorrow), 1 in review, and 1 in draft still.

**Comment by tgolen** [comment]
This is on track to finish soon. Most issues have been closed and just waiting on one last one.

**Comment by trjExpensify** [comment]
👋 Is this done? Moving it to #wave-collect polish in the meantime.

**Comment by tgolen** [comment]
Looks like this is all done! 🎉
